tests/integration/ha_tests/test_self_healing.py

```diff
@@ -154,23 +130,90 @@ async def test_restart_db_process(
+    wal_settings,
+) -> None:
+    """A forceful restart with deleted data and without transaction logs (forced clone)."""
+    # Locate primary unit.
+    app = await app_name(ops_test)
+    primary_name = await get_primary(ops_test, app)
+
+    # Start an application that continuously writes data to the database.
+    await start_continuous_writes(ops_test, app)
+
+    # Copy data dir content removal script.
+    await ops_test.juju(
+        "scp", "tests/integration/ha_tests/clean-data-dir.sh", f"{primary_name}:/tmp"
+    )
+
+    # Stop the systemd service on the primary unit.
+    logger.info(f"stopping database from {primary_name}")
+    await run_command_on_unit(ops_test, primary_name, "/charm/bin/pebble stop postgresql")
+
+    # Data removal runs within a script, so it allows `*` expansion.
+    logger.info(f"removing data from {primary_name}")
+    return_code, _, _ = await ops_test.juju(
+        "ssh",
+        primary_name,
+        "chmod",
+        "+x",
+        "/tmp/clean-data-dir.sh",
+        "&&",
+        "/tmp/clean-data-dir.sh",
+    )
+    assert return_code == 0, "Failed to remove data directory"
+
+    # Wait some time to elect a new primary.
+    sleep(MEDIAN_ELECTION_TIME * 2)
+
+    async with ops_test.fast_forward():
+        await check_writes_are_increasing(ops_test, primary_name)
+
+        # Verify that a new primary gets elected (ie old primary is secondary).
+        for attempt in Retrying(stop=stop_after_delay(60), wait=wait_fixed(3)):
+            with attempt:
+                logger.info("checking whether a new primary was elected")
+                new_primary_name = await get_primary(ops_test, app)
+                assert new_primary_name != primary_name
+
+        # Change some settings to enable WAL rotation.
+        for unit in ops_test.model.applications[app].units:
+            if unit.name == primary_name:
+                continue
+            logger.info(f"enabling WAL rotation on {primary_name}")
+            await change_wal_settings(ops_test, unit.name, 32, 32, 1)
+
+        # Rotate the WAL segments.
+        files = await list_wal_files(ops_test, app)
+        host = await get_unit_address(ops_test, new_primary_name)
+        password = await get_password(ops_test, down_unit=primary_name)
+        with db_connect(host, password) as connection:
+            connection.autocommit = True
+            with connection.cursor() as cursor:
+                # Run some commands to make PostgreSQL do WAL rotation.
+                cursor.execute("SELECT pg_switch_wal();")
+                cursor.execute("CHECKPOINT;")
+                cursor.execute("SELECT pg_switch_wal();")
+        connection.close()
+        new_files = await list_wal_files(ops_test, app)
+        # Check that the WAL was correctly rotated.
+        for unit_name in files:
+            assert not files[unit_name].intersection(
+                new_files
+            ), "WAL segments weren't correctly rotated"
+
```

Comment:
exactly in this place, it is needed to remove all the files which existed before `pg_switch_wal()` on all members.